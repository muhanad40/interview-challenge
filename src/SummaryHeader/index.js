import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  inject,
  observer,
} from 'mobx-react';

import Item from '../Item';

class SummaryHeader extends Component {
  getDietaryCounts() {
    const dietaries = {};

    this.props.previewItems.forEach(item => {
      item.dietaries.forEach(dietaryItem => {
        if (dietaries[dietaryItem] === undefined) {
          dietaries[dietaryItem] = 1;
        } else {
          dietaries[dietaryItem] += 1;
        }
      })
    });

    return dietaries;
  }

  renderDietaryCounts() {
    const counts = this.getDietaryCounts();

    return Object.keys(counts).map(dietaryName => {
      const dietaryCount = counts[dietaryName];

      return (
        <span key={dietaryName}>
          {dietaryCount}x <span className="dietary">{dietaryName}</span>
        </span>
      )
    })
  }

  render() {
    return (
      <div className="menu-summary">
        <div className="container">
          <div className="row">
            <div className="col-6 menu-summary-left">
              <span>{this.props.previewItems.length} items</span>
            </div>

            <div className="col-6 menu-summary-right">
              {this.renderDietaryCounts()}
            </div>
          </div>
        </div>
      </div>
    )
  }
}

SummaryHeader.defaultProps = {
  previewItems: [],
}

SummaryHeader.propTypes = {
  previewItems: PropTypes.arrayOf(PropTypes.shape(Item.propTypes)),
}

export default inject(({
  itemsStore: { previewItems }
}) => {
  return {
    previewItems,
  };
})(observer(SummaryHeader));
