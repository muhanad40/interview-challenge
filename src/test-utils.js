import React from 'react';
import { shallow } from 'enzyme';

export function createRenderer(Component, props) {
  return (extraProps) => shallow(<Component {...props} {...extraProps} />);
}
