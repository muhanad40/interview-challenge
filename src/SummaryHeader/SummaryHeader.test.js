require('../test-setup');

import SummaryHeader from './index';
import { createRenderer } from '../test-utils';

describe('SummaryHeader', () => {
  let render;
  const mockItems = [
    {
      id: 1002,
      name: 'Hake & Smoky Chickpeas, Brown Rice & Quinoa, Roasted Roots',
      dietaries: ['v', 've', 'rsf'],
    },
    {
      id: 1003,
      name: 'Dill & Swiss Chard Potato Cakes, Summer Tabbouleh & Roasted Roots',
      dietaries: ['v', 've'],
    },
    {
      id: 1004,
      name: 'Hake & Smoky Chickpeas, Herby Potatoes & Turmeric Satay Broccoli',
      dietaries: ['rsf'],
    },
    {
      id: 1005,
      name: 'Cuban Beef, Brown Rice & Quinoa, Green Pepper & Butterbean Salad',
      dietaries: ['v', 'rsf'],
    },
  ];
  const initProps = {
    totalCount: mockItems.length,
    items: mockItems,
  };

  beforeAll(() => {
    render = createRenderer(SummaryHeader.wrappedComponent, initProps);
  });

  it('should render correctly', () => {
    expect(render()).toMatchSnapshot();
  });
});
