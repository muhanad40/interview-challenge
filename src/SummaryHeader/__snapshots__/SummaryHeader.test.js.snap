// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryHeader should render correctly 1`] = `
<div
  className="menu-summary"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-6 menu-summary-left"
      >
        <span>
          0
           items
        </span>
      </div>
      <div
        className="col-6 menu-summary-right"
      />
    </div>
  </div>
</div>
`;
